ext/spl/php_spl.c

```diff
@@ -657,18 +657,7 @@ PHP_FUNCTION(spl_object_id)
 
 PHPAPI zend_string *php_spl_object_hash(zend_object *obj) /* {{{*/
 {
-	intptr_t hash_handle, hash_handlers;
-
-	if (!SPL_G(hash_mask_init)) {
-		SPL_G(hash_mask_handle)   = (intptr_t)(php_mt_rand() >> 1);
-		SPL_G(hash_mask_handlers) = (intptr_t)(php_mt_rand() >> 1);
-		SPL_G(hash_mask_init) = 1;
-	}
-
-	hash_handle   = SPL_G(hash_mask_handle)^(intptr_t)obj->handle;
-	hash_handlers = SPL_G(hash_mask_handlers);
-
-	return strpprintf(32, "%016zx%016zx", hash_handle, hash_handlers);
+	return strpprintf(32, "f000000000000000%016zx", (intptr_t)obj->handle);
```

Comment:
This should be the other way around, with the zeros on the right (where previously "handlers" was). Also, why is there an `f` at the start?

Reply:
`f` at start to keep it string if used as an array key.

Reply:
fixed & rebased